docs/architecture/components.mdx

```diff
@@ -92,57 +94,60 @@ The Node Collector is also responsible for recording some k8s metrics, and colle
 
 - Create and manage the `instrumentationconfig.odigos.io` objects
-  - Triggers the `odiglet` to inspect runtime details of the workload.
+  - Triggers `odiglet` to inspect runtime information of the workload.
   - Triggers the scheduler to deploy the Odigos pipeline.
-  - Merge the Odigos instrumentation rules and odigos config to a per-workload configuration to be used by the agent.
+  - Merge the Odigos instrumentation rules and odigos config to a workload scope configuration to be used by the agent.
 - Inject the instrumentation device into the workload manifest resource as virtual device.
-- Monitor the `odigos-enabled` label and cleanup any instrumentation related resources when a source is removed.
+- Monitor the `source.odigos.io` CRD and cleanup any instrumentation related resources when a source is removed.
 
 ### Odiglet
 
-**Type**: The odiglet is a daemon set and will run one pod per node in the cluster
+**Type**: Odiglet is a daemon set and will run one pod per node in the cluster
 
-**K8s Resource**: DaemonSet / `odiglet`
+**K8s Resource**: `kubectl get daemonsets -n odigos-system odiglet`
 
-**Role**: The odiglet component setups the k8s node environment to run the telemetry-collection agents.
+**Role**: The odiglet component sets up the k8s node environment to run the telemetry-collection agents.
 
-- Inspect the runtime details of the workload and detect the programming language, relevant env variables, and runtime versions.
-- Implement the instrumentation devices. On the presence of the supported devices on the pod, odiglet will mount the agent code to the pod, and add the relevant environment variables for starting it and applying the instrumentation.
-- For eBPF based instrumentation, odiglet will inject the eBPF code into the pod, and read the telemetry data from the eBPF queues to push it downstream.
-- Monitor changes to the instrumentation configs, and update the agents accordingly at runtime.
+- Inspect the runtime details of the instrumented sources containers (programming language, relevant env variables, runtime versions, etc).
+- Implement the Kubernetes virtual instrumentation devices. In the presence of a supported devices on the pod, odiglet will mount the agent code from fily-system to the pod, and add the relevant environment variables to initiate instrumentation.
+- For eBPF based instrumentation, odiglet will inject the eBPF code into the pod, pick up the telemetry items from the eBPF queues and push it downstream to node collector component.
+- Monitor changes to instrumentations config, and apply to the relevant agents accordingly at runtime.
 
 ### Node Collector
 
-**Role**: The node collector (in the context of instrumentation) is responsible for collecting k8s metrics, and logs from the node filesystem.
+**Role**: Node collector (in the context of instrumentation) is responsible for scraping Kubernetes metrics, and collect logs from the node filesystem.
```

Comment:
```suggestion
**Role**: Node collector is responsible for scraping Kubernetes metrics, and collecting logs from the node filesystem.
```

is the Node collector ever not in the context of instrumentation?

Reply:
I think same suggestion as above, we can drop the `(in the context of X)`. it's kind of confusing, and these page is about what each component does